components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilderTest.java

```diff
@@ -157,14 +173,80 @@ public void setUp() throws Exception {
 
     }
 
+    @DataProvider(name = "TestBuildIDToken")
+    public Object[][] buildRequestIDToken() throws Exception {
+        // second set of values
+        OAuthTokenReqMessageContext messageContext2 = new OAuthTokenReqMessageContext(tokenReqDTO);
+        messageContext2.setAuthorizedUser(user);
+        messageContext2.setScope(TestConstants.OPENID_SCOPE_STRING.split(" "));
+        messageContext2.addProperty(AUTHORIZATION_CODE, AUTHORIZATION_CODE_VALUE);
+
+        AuthenticatedUser user2 = new AuthenticatedUser();
+        user2.setAuthenticatedSubjectIdentifier(TestConstants.USER_NAME);
+        user2.setUserName(TestConstants.USER_NAME);
+        user2.setUserStoreDomain(TestConstants.USER_STORE_DOMAIN);
+        user2.setTenantDomain(SUPER_TENANT_DOMAIN_NAME);
+        user2.setFederatedUser(true);
+        messageContext2.setAuthorizedUser(user2);
+
+        return new Object[][]{
+                {messageContext, tokenRespDTO},
+                {messageContext2, tokenRespDTO}
+        };
+    }
+
+    @Test(dataProvider = "TestBuildIDToken")
+    public void testBuildIDToken(OAuthTokenReqMessageContext messageContext,
+                                 OAuth2AccessTokenRespDTO tokenRespDTO) throws Exception {
+
+        RealmService realmService = IdentityTenantUtil.getRealmService();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                .setUserRealm(realmService.getTenantUserRealm(SUPER_TENANT_ID));
+        IdpMgtServiceComponentHolder.getInstance().setRealmService(IdentityTenantUtil.getRealmService());
+        String idToken = defaultIDTokenBuilder.buildIDToken(messageContext, tokenRespDTO);
+        Object o = SignedJWT.parse(idToken).getJWTClaimsSet();
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getAudience().get(0),
+                TestConstants.CLIENT_ID);
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getIssuer(),
+                "https://localhost:9443/oauth2/token");
+        }
+
     @Test
-    public void testBuildIDToken() throws Exception {
+    public void testAuthoBuildIDToken() throws Exception {
```

Comment:
```suggestion
    public void testAuthoBuildIDToken() throws Exception {
```
testBuildIDToken(OAuthAuthzReqMessageContext authzReqMessageContext,
                               OAuth2AuthorizeRespDTO tokenRespDTO)

Use dataproviders and pass different objects for authzReqMessageContext and tokenRespDTO in order to improve the test